src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate localDateTo = LocalDate.parse(dateTo, formatter);
+
+        for (String name : names) {
+            int salary = 0;
+
+            for (String employeeData : data) {
+                String[] parts = employeeData.split(" ");
+                final int dateIndex = 0;
+                final int nameIndex = 1;
+                final int workingHoursIndex = 2;
+                final int salaryPerHourIndex = 3;
```

Comment:
It is great, but it is better to do them on class level and do as constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)